src/preprocessing/passes/sygus_inference.cpp

```diff
@@ -197,10 +197,13 @@ bool SygusInference::solveSygus(std::vector<Node>& assertions,
         }
         else if (cur.isVar() && cur.getKind() != BOUND_VARIABLE)
         {
-          // a free variable is a 0-argument function to synthesize
-          Assert(std::find(free_functions.begin(), free_functions.end(), cur)
-                 == free_functions.end());
-          free_functions.push_back(cur);
+          // A free variable that is not in a higher-order context is a
+          // 0-argument function-to-synthesize.
```

Comment:
Please clarify that you can ignore the HO functions here because that are accounted for in a different part of the code

Reply:
I'm not ignoring them, I'm ensuring we don't add duplicates to the vector.

Reply:
Yes, you are ignoring the duplicate. If one looks only to the comment one can be confused, as I was, that they are duplicates. It'd be to say why they are duplicates.

Reply:
I've made the fix more robust, see my latest commit.